### PR TITLE
Add automatic release creation

### PR DESCRIPTION
**Pull request by nckturner**
**What this PR does / why we need it**:

With this change, we will run goreleaser on tag creation to automatically create a release with binaries.  We are also reducing build duplication by remove `Dockerfile.goreleaser` and the docker image build in goreleaser.  Only `Dockerfile` will be used and we will no longer be building images and pushing the 602401143452 account which relied on a manual build step.  Instead, we will be using cloudbuild and promoting images via the CNCF artifact publishing machinery. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

Test release generated here: https://github.com/nckturner/aws-iam-authenticator/releases/tag/v9.9.10 (The admin still needs to click publish after goreleaser runs).

### CODE REVIEW

File: .github/workflows/create-release.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        with:
+          fetch-depth: 0
+      -
+        name: Set up Go
+        uses: actions/setup-go@v3
+      -
+        name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v4
+        with:
+          distribution: goreleaser
```
Comment (nnmin-aws) [inline .github/workflows/create-release.yml:25]: this is awesome! 
Could you please kindly add a comment here or in Readme.md where to locate the new release image?

Comment (nckturner) [reply .github/workflows/create-release.yml:25]: Yes, I'll add something to the README.md but if its ok I'll do it in a follow up PR.

File: .github/workflows/create-release.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          version: latest
```
Comment (jaypipes) [inline .github/workflows/create-release.yml:26]: Do we have a process for using goreleaser for non-latest tags?

Comment (jaypipes) [reply .github/workflows/create-release.yml:26]: Never mind... I think this `latest` is referring to *goreleaser*, is that right?

Comment (nckturner) [reply .github/workflows/create-release.yml:26]: Yes, correct, just saying use the latest goreleaser binary... which maybe we should change.

**Pull request merged**